cas-server-core-tickets/src/main/java/org/jasig/cas/web/flow/TicketExceptionHandler.java

```diff
@@ -0,0 +1,95 @@
+package org.jasig.cas.web.flow;
+
+import com.google.common.collect.ImmutableList;
+import org.jasig.cas.ticket.AbstractTicketException;
+import org.jasig.cas.ticket.UnsatisfiedAuthenticationPolicyException;
+import org.springframework.binding.message.MessageBuilder;
+import org.springframework.binding.message.MessageContext;
+import org.springframework.stereotype.Component;
+
+import javax.validation.constraints.NotNull;
+import java.util.List;
+
+/**
+ * Performs two important error handling functions on an {@link AbstractTicketException} raised from the login webflow
+ * layer:
+ *
+ * <ol>
+ *     <li>Maps ticket errors onto message bundle strings for display to user for configured errors only.</li>
+ *     <li>Determines the next webflow state by comparing ticket errors against {@link #errors} in list order. The first
+ *     entry that matches determines the outcome state, which is the simple class name of the exception.</li>
+ * </ol>
+ *
+ * @author Daniel Frett
+ * @since 4.3.0
+ */
+@Component("ticketExceptionHandler")
+public class TicketExceptionHandler {
```

Comment:
This is almost identical to the existing ticket exception handler. I think we should rework that component into this one, or vice versa to remove duplicate/code and logic. I suspect that you should be able to define your mappings directly in that handler already, no?